fe/fe-core/src/main/java/com/starrocks/load/routineload/RoutineLoadTaskScheduler.java

```diff
@@ -246,13 +246,23 @@ private void scheduleOneTask(RoutineLoadTaskInfo routineLoadTaskInfo) throws Exc
             LOG.debug("create routine load task cost(ms): {}, job id: {}",
                     (System.currentTimeMillis() - startTime), routineLoadTaskInfo.getJobId());
         } catch (MetaNotFoundException e) {
-            releaseBeSlot(routineLoadTaskInfo);
-            // this means database or table has been dropped, just stop this routine load job.
-            routineLoadManager.getJob(routineLoadTaskInfo.getJobId())
-                    .updateState(JobState.CANCELLED,
-                            new ErrorReason(InternalErrorCode.META_NOT_FOUND_ERR, "meta not found: " + e.getMessage()),
-                            false);
-            throw e;
+            if ((e.getMessage().startsWith("database ") || e.getMessage().startsWith("table "))
+                    && e.getMessage().endsWith(" does not exist")) {
```

Comment:
it is not good to distinguish by error message.
how about throw `LoadException` or `UserException` when txn is not found?